matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/keysbackup/RustKeyBackupService.kt

```diff
@@ -822,116 +799,110 @@ internal class RustKeyBackupService @Inject constructor(
     /**
      * Do a backup if there are new keys, with a delay
      */
-    fun maybeBackupKeys() {
-        when {
-            isStucked                              -> {
-                // If not already done, or in error case, check for a valid backup version on the homeserver.
-                // If there is one, maybeBackupKeys will be called again.
-                checkAndStartKeysBackup()
-            }
-            state == KeysBackupState.ReadyToBackUp -> {
-                keysBackupStateManager.state = KeysBackupState.WillBackUp
-
-                // Wait between 0 and 10 seconds, to avoid backup requests from
-                // different clients hitting the server all at the same time when a
-                // new key is sent
-                val delayInMs = Random.nextLong(KEY_BACKUP_WAITING_TIME_TO_SEND_KEY_BACKUP_MILLIS)
-
-                cryptoCoroutineScope.launch {
-                    delay(delayInMs)
-                    // TODO is this correct? we used to call uiHandler.post() instead of this
-                    withContext(Dispatchers.Main) {
-                        backupKeys()
+    suspend fun maybeBackupKeys() {
+        withContext(coroutineDispatchers.crypto) {
+            when {
+                isStucked                              -> {
```

Comment:
Can we rename this, stucked isn't a word :sweat_smile:. 

I didn't do this myself because it's part of the `KeysBackupService` which seems to be a public interface, but you seem to have already introduced changes to them.